src/main/java/com/minecolonies/coremod/network/messages/BuildToolPlaceMessage.java

```diff
@@ -146,7 +146,7 @@ public void messageOnServerThread(final BuildToolPlaceMessage message, final Ent
      */
     private static void handleHut(
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 27 to the 20 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)